modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -224,6 +224,15 @@ struct IEUnit {
 
     explicit IEUnit(const cv::gapi::ie::detail::ParamDesc &pp)
         : params(pp) {
+        InferenceEngine::ParamMap ctx_params;
+        try {
+            ctx_params = cv::util::any_cast<InferenceEngine::ParamMap>(params.context_config);
+            auto ie_core = cv::gimpl::ie::wrap::getCore();
+            params.rctx = ie_core.CreateContext(params.device_id, ctx_params);
+        } catch (...) {
+            // Do nothing here
+        }
```

Comment:
This may be not the best way to handle the situation when the underlying type is different.

`catch/try` is evil for this purpose, maybe you need to use `any_cast(*)` (takes a pointer and returns a pointer, maybe nullptr).

Reply:
Done, please, take a look

Reply:
So it seems you don't even need to have `.rctx` in the params - instead, you shold have it right in your `IEUnit`?

Reply:
`rctx` and the params are used to load/import network for example, it cannot be moved to `IEUnite`

Reply:
Moved to IEUnit